src/typescript/index.d.ts

```diff
@@ -0,0 +1,430 @@
+declare module "web3-eea" {
+    import type Web3 from "web3";
+    import type { TransactionConfig } from "web3-eth";
+    import type BN from "bn.js";
+
+    export default function EEAClient(web3: Web3, chainId: number): IWeb3InstanceExtended;
+
+    export interface IEeaWeb3Extension {
+        /**
+         * Send the Raw transaction to the Besu node
+         */
+        sendRawTransaction: (options: ISendRawTransactionOptions) => Promise<string>;
+    }
+
+    export interface IWeb3InstanceExtended extends Web3 {
+        eea: IEeaWeb3Extension;
+        priv: IPrivWeb3Extension;
+        privx: IPrivxWeb3Extension;
+    }
+
+    export interface IPrivxWeb3Extension {
+        /**
+         * Creates an on chain privacy group
+         */
+        createPrivacyGroup(options: ICreatePrivacyGroupOptions): Promise<IPrivateTransactionReceipt>;
+        /**
+         * @deprecated
+         */
+        findOnChainPrivacyGroup(options: IFindOnChainPrivacyGroupOptions): Promise<unknown>;
+        /**
+         * Remove a member from an on-chain privacy group
+         */
+        removeFromPrivacyGroup(options: IRemoveFromPrivacyGroupOptions): Promise<IPrivateTransactionReceipt>;
+        /**
+         * Add to an existing on-chain privacy group
+         */
+        addToPrivacyGroup(options: IAddToPrivacyGroupOptions): Promise<IPrivateTransactionReceipt>;
+        /**
+         * Either lock or unlock the privacy group for member adding
+         */
+        setPrivacyGroupLockState(options: ISetPrivacyGroupLockState): Promise<IPrivateTransactionReceipt>;
+    }
+    export interface IPrivWeb3Extension {
+        /**
+         * Generate a privacyGroupId
+         * @param options Options passed into `eea_sendRawTransaction`
+         * @returns String The base64 encoded keccak256 hash of the participants.
+         */
+        generatePrivacyGroup(options: IGeneratePrivacyGroupOptions): string;
+        /**
+         * Returns with the deleted group's ID (same one that was passed in).
+         */
+        deletePrivacyGroup(options: IDeletePrivacyGroupOptions): Promise<string>;
+        /**
+         * Returns a list of privacy groups containing only the listed members.
+         * For example, if the listed members are A and B, a privacy group
+         * containing A, B, and C is not returned.
+         */
+        findPrivacyGroup(options: IFindPrivacyGroupOptions): Promise<Array<IPrivacyGroup>>;
+        distributeRawTransaction(options: IPrivateTransactionConfig, method: string): Promise<unknown>;
```

Comment:
there is no `method` param

Reply:
@achraf17 Done: https://github.com/petermetz/web3js-eea/commit/740a32174f1f00f0d6a32caf0edd1fc274168739